src/ui/layer_bar.ts

```diff
@@ -168,6 +190,21 @@ class LayerWidget extends RefCounted {
     }
     title += ", drag to move, shift+drag to copy";
     element.title = title;
+    // Color widget updates
+    if (panel.layerGroupViewer.viewerState.enableLayerColorWidget.value) {
+      if (layer.supportsLayerBarColorSyncOption) {
+        const color = this.layer.layerBarColor;
+        if (color) {
+          element.dataset.color = "fixed";
+          layerColorElement.style.backgroundColor = color;
+        } else {
+          element.dataset.color = "rainbow";
+        }
+      } else {
+        element.dataset.color = "unsupported";
+      }
+    }
+    layerColorElement.title = layer.colorWidgetTooltip() || "";
```

Comment:
I think we should give non-supported layers a message, something like
`layerColorElement.title = layer.colorWidgetTooltip() || "The color of this layer cannot be determined";`